poseidon/src/lib.rs

```diff
@@ -258,7 +258,7 @@ pub fn hashv(
     {
         let mut hash_result = [0; HASH_BYTES];
         let result = unsafe {
-            crate::syscalls::sol_poseidon(
+            solana_program::syscalls::sol_poseidon(
```

Comment:
This appears to be the only reason this crate needs to depend on `solana-program` at all.

I'm not 100% sure this is The Way™, but we could just implement this syscall binding manually in the crate.

```rust
extern "C" {
    pub fn sol_poseidon(
        parameters: u64,
        endianness: u64,
        vals: *const u8,
        val_len: u64,
        hash_result: *mut u8,
    ) -> u64;
}
```

Then this line just becomes

```suggestion
            sol_poseidon(
```

@CriesofCarrots would love your thoughts on this.

Reply:
Sounds fine to me anyway!